nbs/04b_doclinks.ipynb

```diff
@@ -33,7 +33,7 @@
     "from fastcore.utils import *\n",
```

Comment:
<p>Is this needed for this particular PR? AFAICT it's just to avoid some possible problem where the marker is inside a multi-line string?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/fastai/nbdev/pull/928/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Reply:
<p>It's needed because rewriting the source code from the ast in this method: https://github.com/fastai/nbdev/blob/master/nbdev/maker.py#L209 causes some reformatting from a single line to a multiline and then it's treated as a separate notebook cell. I was thinking of submitting this as a separate patch, but it's not needed unless someone changes the formatting of that file.</p>

Reply:
I'd rather avoid adding such complex code unless it's definitely required! And if we really do need it, we should include plenty of comments explaining what it's doing and why. Or better still, find a way to avoid it.

Reply:
<p>Yes, the error raised by this was a bit of head scratcher and I had to do a bit of digging on <code>ast</code> and newlines. As an example: https://stackoverflow.com/questions/53636723/python-parsing-code-with-new-line-character-in-them-using-ast.</p><p> </p><p>This is only an issue if:</p><ol><li>someone tries to rewrite that line as a triple quoted string using actual newlines</li><li>anything happens to parse that method using ast and rewrite it (which happens here since it has the @patch annotation). </li></ol><p> </p><p>I can guarantee you that I had exactly zero fun figuring out this regex, and I only did it to make progess on the actual problem that I wanted to solve (making @patch more usable). I'll leave it up to you to decide.</p>